pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -237,7 +249,9 @@ func NewDeploymentController(client kube.Client, clusterID cluster.ID,
 	// On injection template change, requeue all gateways
 	injectionHandler(func() {
 		for _, gw := range dc.gateways.List(metav1.NamespaceAll, klabels.Everything()) {
-			dc.queue.AddObject(gw)
+			if string(gw.Spec.GatewayClassName) == gw.GetName() {
```

Comment:
why are we comparing the GWC to the GW name?

Reply:
This was copy pasta.  Fixing now.